### PR TITLE
subscription leaks removed

### PR DESCRIPTION
**Pull request by jydimir**
Hi!

Second and following calls to `MainPresenter.loadRepos()` probably will cause the subscriptions leaks, because you don't unsubscribe from previous and lose reference to them. One quick suggestion:  unsubscribe from previous subscription before doing `mSubscription = mDataManager.getRepository()...`.

### CODE REVIEW

**Comment by ribotaccounts** [comment]
Can one of the admins verify this patch?

File: app/src/main/java/uk/co/ribot/androidboilerplate/util/RxUtil.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package uk.co.ribot.androidboilerplate.util;
+
+import rx.Subscription;
+
+public class RxUtil {
+
+    public static void unsubscribe(Subscription subscription) {
+        if((subscription != null) && (!subscription.isUnsubscribed())) {
```
Comment (ivacf) [inline app/src/main/java/uk/co/ribot/androidboilerplate/util/RxUtil.java]: There are a few unnecessary brackets I believe.
Also there should be a space between the if and the first bracket:
```java
if (subscription != null && !subscription.isUnsubscribed()) {
```

**Comment by ivacf** [comment]
Yeah, that's another option. If we use `CompositeSubscription` we could end up having more than one subscription running on parallel. If for some reason this method is called multiple times (that it shouldn't really happen) I think it's probably better to cancel the previous subscription and trigger a new one?

**Comment by jydimir** [comment]
@digitalbuddha @ivacf Yeah definitely, `CompositeSubscription.add(...)` used in  `loadRibots()` method will cause mutiple subscriptions running on parallel. 

Canceling previous subscription and creating new one(suggested in this pull request) is one of two possible solutions. Another solution is to use `CompositeSubscription` and `PublishSubject`. I always do this in more complicated cases. But for this simple boilerplate I think is better to use "cancel and create" way.

**Comment by ivacf** [comment]
Thanks

**Pull request merged**